docs/book/developer-guide/materializer.md

```diff
@@ -155,13 +150,13 @@ class MyMaterializer(BaseMaterializer):
 Pro-tip: Use the ZenML `fileio` handler to ensure your materialization logic works across artifact stores (local and 
 remote like S3 buckets).
 
-Then edit the pipeline as follows:
+Now ZenML can use this materializer to handle outputs and inputs of your customs object. Edit the pipeline as follows 
+to see this in action:
 
 ```python
-pipe(
-    step1=step1().with_return_materializers(MyMaterializer),
-    step2=step2()
-).run()
+first_pipeline(
+    step_1=my_first_step().with_return_materializers(MyMaterializer),
```

Comment:
I'm not 100% sure, but how I understand it you wouldn't even need to specify the `with_return_materializer` right?  As the class `MyObj` is already an associated type of the materializer, it should automatically detect it. Not sure if we should mention that functionality somehow

Reply:
Very good question, I'll try to find a way of putting a sentence in there about this. I feel Materializers are the most misunderstood part of ZenML atm, so it makes sense to put some elbow grease in here